clang/test/Driver/darwin-header-search-libcxx.cpp

```diff
@@ -172,3 +172,32 @@
 // RUN:               --check-prefix=CHECK-LIBCXX-STDLIB-UNSPECIFIED %s
 // CHECK-LIBCXX-STDLIB-UNSPECIFIED: "-cc1"
 // CHECK-LIBCXX-STDLIB-UNSPECIFIED: "-internal-isystem" "[[SYSROOT]]/usr/include/c++/v1"
+
+// ----------------------------------------------------------------------------
```

Comment:
Can you try to integrate your test to the existing ones a bit more closely? They are not really specific to xpm/npm, so it makes sense to support them in a platform agnostic way.

For example, we need to add a test where we set up a symlinked Clang and then run with a toolchain that has headers in it, and confirm that this takes precedence over the ones in the symlink directory. In other words, we now have (1), (2) and (3) places where we look for headers in. Previously, we pretty exhaustively checked the combinations for the two locations we checked -- now we should at least try to check:

- Headers in (1) and in (2) -> (1) is preferred over (2)
- Headers in (2) and in (3) -> (2) is preferred over (3) -- you already check that with your current test
- Headers in (2) and nowhere else -> (2) is used

Also, we should make the comments in the tests platform-agnostic. We also probably want to insert the tests in the right place in the file and I suspect that using `(1)` `(2)` and `(3)` terminology to explain what we're testing would help.

Reply:
BTW this looks good to me apart from these test improvements.